### DISCUSSION TITLE
This conditional is hypothetically unreachable..

### DISCUSSION THREAD

**Discussion by YavorJJ** [Cyfrin/foundry-full-course-cu#1971] [Q&A] [answered]
Can some help with Patrick's comment here, i rewatched it couple of times Patrick's explaining it, but I can't grasp it. Thank you.
![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/171678195/2690b441-8205-4ce1-b146-215171d5708f)
This is the comment: This conditional is hypothetically unreachable
Defi StableCoin Lesson 13.

**Comment by EngrPips** [comment, marked as answer]
Hello @YavorJJ, If you are burning `DSC,` your health factor will increase, meaning you will take off your debt and give your collateral more breathing space. So if you burn some `DSC,` the `DSCEngine` would take some `DSC` token from you, and there is no possibility of that process failing, assuming you are burning some `DSC` that you have initially minted. That's how I understood the comment. I am not sure if that makes sense to you.

**Reply by YavorJJ** [reply]
It makes sense, thank you

**Reply by EngrPips** [reply]
Awesome.
